### PR TITLE
fix(hash): updated sha1 with sha256 for file hash

### PR DESCRIPTION
**Pull request by cnlangzi**
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/blobber/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master BEFORE merging this PR

Associated PRs (Link as appropriate):
- zbox: https://github.com/0chain/zboxcli/pull/159
- zwalelet:
- blobber: https://github.com/0chain/blobber/pull/515
- 0chain:
- system_test: https://github.com/0chain/system_test/pull/173

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/update_sha1_with_sha256_for_file_hash  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | pr-515  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/update_sha1_with_sha256_for_file_hash  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/update_sha1_with_sha256_for_file_hash  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/fix_thumbnail_assertion  |
| gosdk  | fix/update_sha1_with_sha256_for_file_hash  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/update_sha1_with_sha256_for_file_hash  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by kushthedude — COMMENTED** [review]
This looks good but we would need to upgrade the zboxcli & blobber to utilise this gosdk version & then re-run tests.
I guess this PR should be the first to go into staging & later the blobber then zboxcli.

File: dev/blobber/model/write_marker.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+type WriteMarker struct {
+	AllocationRoot         string           `gorm:"column:allocation_root;primary_key" json:"allocation_root"`
+	PreviousAllocationRoot string           `gorm:"column:prev_allocation_root" json:"prev_allocation_root"`
+	AllocationID           string           `gorm:"column:allocation_id" json:"allocation_id"`
+	Size                   int64            `gorm:"column:size" json:"size"`
+	BlobberID              string           `gorm:"column:blobber_id" json:"blobber_id"`
+	Timestamp              common.Timestamp `gorm:"column:timestamp" json:"timestamp"`
+	ClientID               string           `gorm:"column:client_id" json:"client_id"`
+	Signature              string           `gorm:"column:signature" json:"signature"`
+}
```
Comment (kushthedude) [inline dev/blobber/model/write_marker.go:14]: Are we planning to abstract away models into seperate package?

**Review by kushthedude — APPROVED** [review]

**Comment by cnlangzi** [comment]
> This looks good but we would need to upgrade the zboxcli & blobber to utilise this gosdk version & then re-run tests. I guess this PR should be the first to go into staging & later the blobber then zboxcli.

all relates PRs are passed on system_test. blobber will calculate and return hash to gosdk/zbox, client will verify it again. so sha256 should be changed on blobber + gosdk + zbox at the same time. and all changes art listed on related PR links.

File: dev/blobber/model/write_marker.go (hunk 1 above)
Comment (cnlangzi) [reply dev/blobber/model/write_marker.go:14]: it is a compact mock object, just is used on mock server now.

**Review by ssardana96 — APPROVED** [review]

**Comment by ssardana96** [comment]
lgtm
I guess all 3 needs to be merged in sequence to have all the tests pass. considering test should be ok, changes looks good!

**Pull request merged**